src/tree.rs

```diff
@@ -1,46 +1,24 @@
 #[derive(Debug)]
-pub enum Tree {
-    Leaf {
-        id: u64,
-        name: String,
-        contents: String,
-    },
-    Node {
-        id: u64,
-        name: String,
-        children: Vec<Tree>,
-    },
+pub enum Contents {
+    Leaf(String),
+    Node(Vec<Tree>),
 }
 
-impl Tree {
-    pub fn id(&self) -> u64 {
-        match self {
-            Tree::Leaf { id, .. } => *id,
-            Tree::Node { id, .. } => *id,
-        }
-    }
-
-    pub fn name(&self) -> &str {
-        match self {
-            Tree::Leaf { name, .. } => name,
-            Tree::Node { name, .. } => name,
-        }
-    }
-
-    pub fn name_mut(&mut self) -> &mut String {
-        match self {
-            Tree::Leaf { name, .. } => name,
-            Tree::Node { name, .. } => name,
-        }
-    }
+#[derive(Debug)]
+pub struct Tree {
+    pub(crate) id: u64,
+    pub(crate) name: String,
+    pub(crate) contents: Contents,
```

Comment:
Currently this crate doesn't have a library, so `pub` is the same as `pub(crate)`. That's why I made everything `pub` that needs is used in another module. Maybe do that too here? TIOLI!

```suggestion
    pub id: u64,
    pub name: String,
    pub contents: Contents,
```